shared/chat/conversation/info-panel/index.js

```diff
@@ -44,183 +39,222 @@ type InfoPanelProps = {
-            teamname={props.teamname}
-            participantCount={props.participantCount}
-            onClick={props.onViewTeam}
-          />
-
-          <Divider style={{marginBottom: 20, marginTop: 20}} />
-
-          <Notifications />
-          <Divider style={dividerStyle} />
-
-          <ManageTeam
-            canManage={props.admin}
-            label="In this team"
-            participantCount={props.participantCount}
-            onClick={props.onViewTeam}
-          />
-        </Box>
+        <BigTeamHeader
+          key="big team header"
+          channelname={row.channelname}
+          teamname={row.teamname}
+          onClick={row.onViewTeam}
+        />
       )
 
-    case 'big header':
+    case 'join channel':
       return (
-        <Box key={props.key} style={{...globalStyles.flexBoxColumn, alignItems: 'stretch'}}>
-          <BigTeamHeader
-            channelname={props.channelname}
-            teamname={props.teamname}
-            onClick={props.onViewTeam}
-          />
-
-          {!props.isPreview && (
-            <Box>
-              <Divider style={dividerStyle} />
-              <Notifications />
-            </Box>
-          )}
-
-          <Divider style={dividerStyle} />
-
-          <Box style={{...globalStyles.flexBoxRow, justifyContent: 'center'}}>
-            {props.isPreview && (
-              <Button
-                type="Primary"
-                label="Join channel"
-                style={{marginRight: globalMargins.xtiny}}
-                small={true}
-                onClick={props.onJoinChannel}
-              />
-            )}
-            {!props.isPreview && (
-              <Button type="Danger" small={true} label="Leave channel" onClick={props.onLeaveConversation} />
-            )}
-          </Box>
-
-          {props.isPreview && (
-            <Text type="BodySmall" style={{textAlign: 'center', marginTop: globalMargins.xtiny}}>
-              Anyone in {props.teamname} can join.
-            </Text>
-          )}
-
-          <Divider style={dividerStyle} />
-
-          <ManageTeam
-            canManage={props.admin && props.channelname === 'general'}
-            label="In this channel"
-            participantCount={props.participantCount}
-            onClick={props.onViewTeam}
-          />
-        </Box>
+        <CaptionedButton
+          caption={`Anyone in ${row.teamname} can join.`}
+          key="join channel"
+          label="Join channel"
+          onClick={row.onJoinChannel}
+        />
       )
-    case 'participant':
-      return <Participant key={props.key} {...props} />
+
+    case 'leave channel':
+      return <DangerButton key="leave channel" label="Leave channel" onClick={row.onLeaveConversation} />
 
     default:
-      throw new Error('Unexpected type ' + props.type)
+      // eslint-disable-next-line no-unused-expressions
+      ;(row.type: empty)
+      throw new Error(`Impossible case encountered: ${row.type}`)
```

Comment:
might be good to add some context here since we don't get filenames in logs (😓)

Reply:
actually I'm not sure we get this text in logs anyway?

Reply:
Hmm, this will black bar right? I think it's fine, since this is only to work around flow's inability to infer that this branch isn't taken (from the check above), and if we do hit this at run time there are only a couple of places to check.